neat-screen.js

```diff
@@ -84,155 +83,217 @@ function NeatScreen (cabal) {
-            if (!self.state.user) updateLocalKey()
+      cabal.on('peer-added', function (key) {
+        var found = false
+        Object.keys(cabal.client.users).forEach(function (k) {
+          if (k === key) {
+            cabal.client.users[k].online = true
+            found = true
+          }
+        })
+        if (!found) {
+          cabal.client.users[key] = {
+            key: key,
+            online: true
+          }
+        }
+        self.bus.emit('render')
+      })
+      cabal.on('peer-dropped', function (key) {
+        Object.keys(cabal.client.users).forEach(function (k) {
+          if (k === key) {
+            cabal.client.users[k].online = false
             self.bus.emit('render')
-          })
+          }
         })
+      })
 
-        function updateLocalKey () {
-          self.cabal.getLocalKey(function (err, lkey) {
-            if (err) return self.bus.emit('render')
-            Object.keys(users).forEach(function (key) {
-              if (key === lkey) {
-                self.state.user = users[key]
-                self.state.user.local = true
-                self.state.user.online = true
-                self.state.user.key = key
-              }
-            })
-            self.bus.emit('render')
+      function updateLocalKey () {
+        cabal.getLocalKey(function (err, lkey) {
+          if (err) return self.bus.emit('render')
+          Object.keys(users).forEach(function (key) {
+            if (key === lkey) {
+              cabal.client.user = users[key]
+              cabal.client.user.local = true
+              cabal.client.user.online = true
+              cabal.client.user.key = key
+            }
           })
-        }
-      })
+          self.bus.emit('render')
+        })
+      }
     })
   })
 }
 
+NeatScreen.prototype.showCabal = function (cabal) {
+  this.state.cabal = cabal
+  this.commander.cabal = cabal
+  this.loadChannel(this.state.cabal.client.channel)
+  this.bus.emit('render')
+}
+
 function renderApp (state) {
   if (process.stdout.columns > 80) return views.big(state)
   else return views.small(state)
 }
 
 // use to write anything else to the screen, e.g. info messages or emotes
 NeatScreen.prototype.writeLine = function (line) {
-  this.state.messages.push(`${chalk.gray(line)}`)
+  this.state.cabal.client.messages.push(`${chalk.gray(line)}`)
   this.bus.emit('render')
 }
 
 NeatScreen.prototype.clear = function () {
-  this.state.messages = []
+  this.state.cabal.client.messages = []
   this.bus.emit('render')
 }
 
 NeatScreen.prototype.loadChannel = function (channel) {
-  if (this.state.msgListener) {
-    this.cabal.messages.events.removeListener(this.state.channel, this.state.msgListener)
-    this.state.msgListener = null
-  }
-
   var self = this
+  if (self.state.cabal.client.msgListener) {
+    self.state.cabal.messages.events.removeListener(self.state.cabal.client.channel, self.state.cabal.client.msgListener)
```

Comment:
it's feeling like we'll need to refactor this in the future, it's a very long sequence of characters haha

Reply:
Yeah! Perhaps that will come when we add in cabal-client